python/pip_install/pip_repository.bzl

```
@@ -424,6 +431,15 @@ python_interpreter. An example value: "@python3_x86_64-unknown-linux-gnu//:pytho
 Prefix for the generated packages will be of the form `@<prefix><sanitized-package-name>//...`
 """,
     ),
+    "requirement_groups": attr.string_list_dict(
+        default = {},
+        doc = """\
+A mapping of requirements which form dependency cycles into groups.
+
+Groups of packages will be wrapped so that if a dependency is taken on a member of the cycle the rest of the cycle will
+be correctly included as transitive dependencies.
```

It would be nice to improve the docs to be more user friendly.

```suggestion
A mapping of dependency cycle names to the list of packages that form a cycle.

This attribute allows `rules_python` to work around cycles in Python dependency graph. If you see a dependency cycle like
```
EXAMPLE error
```

Then you can include the following key, value pair to resolve the error:
```
EXAMPLE
```

Note that the key of this dictionary is arbitrary
and the recommendation is to use a value that is easy for humans to debug when looking at dependency graphs.

Adding a group of packages here will ensure that
all of the packages in the dependency cycle are included in the transitive dependency list when any of them is used as dependencies in your code or third party Python libraries.
```